### PR TITLE
PW-673 Implement Secured Fields part2

### PR DESCRIPTION
**Pull request by maassenbas**

### CODE REVIEW

File: cartridges/LINK_adyen/cartridge/client/default/js/adyenSecureFields.js [hunk 1]
```diff
@@ -17,25 +18,36 @@
...
             onConfigSuccess: function() {}, // Called once the secured fields are ready to use
             onFieldValid : function() {}, // Called as a specific secured field is validated and encrypted.
-            onBrand: function() {}, // Called once we detect the card brand
-            onError: function() {}, // Called in the case of invalid card number / invalid expiry date / incomplete field.
+            onBrand: function(brandObject) {
+                $('#cardType').val(brandObject.brand);
+            }, // Called once we detect the card brand
+            onError: function() {
+
+            }, // Called in the case of invalid card number / invalid expiry date / incomplete field.
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/client/default/js/adyenSecureFields.js]: as this is not implemented we can remove it.

File: cartridges/LINK_adyen/cartridge/client/default/js/adyenSecureFields.js [hunk 2]
```diff
@@ -17,25 +18,36 @@
...
             onFocus: function() {}, // Called when a secured field gains or loses focus
-            onBinValue: function(bin) {} // Provides the BIN Number of the card (up to 6 digits), called as the user types in the PAN
+            onBinValue: function(bin) {
+                console.log(bin);
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/client/default/js/adyenSecureFields.js]: remove the console.log.

File: cartridges/LINK_adyen/cartridge/controllers/CheckoutServices.js [hunk 3]
```diff
@@ -6,37 +6,28 @@ server.extend(module.superModule);
...
+      viewData.adyenEncryptedExpiryMonth = paymentForm.creditCardFields.adyenEncryptedExpiryMonth.value;
+      viewData.adyenEncryptedExpiryYear = paymentForm.creditCardFields.adyenEncryptedExpiryYear.value;
+      viewData.adyenEncryptedSecurityCode = paymentForm.creditCardFields.adyenEncryptedSecurityCode.value;
+
+      viewData.paymentInformation = {
+            cardType: {
+              value: paymentForm.creditCardFields.cardType.value
+            },
+            cardNumber: {
+              value: paymentForm.creditCardFields.cardNumber.value
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/CheckoutServices.js]: is this the masked cardNumber ?

File: cartridges/LINK_adyen/cartridge/client/default/js/adyenSecureFields.js [hunk 4]
```diff
@@ -17,25 +18,36 @@
...
+        card = checkout.create('card', {
             // Mandatory fields
             originKey: originKey,
             loadingContext: loadingContext, // The environment where we should loads the secured fields from
             type: 'card',
 
             // Events
-            onChange: function() {}, // Gets triggered whenever a user changes input
-            onValid : function() {}, // Gets triggered when all fields are valid.
+            onChange: function() {
```
Comment (cyattilakiss) [inline cartridges/LINK_adyen/cartridge/client/default/js/adyenSecureFields.js]:
```suggestion
            onChange: function(state) {
                isValid = state.isValid
```

File: cartridges/LINK_adyen/cartridge/forms/default/creditCard.xml [hunk 5]
```diff
@@ -1,19 +1,13 @@
...
-        mandatory="false"
-        max-length="16"/>
-
     <field formid="paymentMethod" type="string" mandatory="true" />
 
     <field formid="cardType" label="profile.cardtype" type="string" mandatory="false" binding="cardType"
            max-length="16" missing-error="error.message.required" />
 
+    <field formid="cardOwner" label="label.input.creditcard.owner" type="string" mandatory="false" binding="cardOwner"
+           max-length="16" missing-error="error.message.required" />
```
Comment (Aleffio) [inline cartridges/LINK_adyen/cartridge/forms/default/creditCard.xml]: let's set max-length to 22 which is the maximum for Mastercard

File: cartridges/LINK_adyen/cartridge/controllers/CheckoutServices.js (hunk 3 above)
Comment (maassenbas) [reply cartridges/LINK_adyen/cartridge/controllers/CheckoutServices.js]: Yes, this is only the BIN. 
Will create masked card number once LastFour is available in callback.

**Pull request merged**